File: BE/common-service/src/main/java/com/manager/common/infrastructure/utils/DateUtils.java
package com.manager.common.infrastructure.utils;

import java.text.SimpleDateFormat;
import java.util.Date;

public class DateUtils {
    public static final String DD_MM_YYYY_HH_MM_SS = "dd/MM/yyyy HH:mm:ss";

    public static String convertToString(Date date, String format) {
        if (date == null) return null;
        SimpleDateFormat sdf = new SimpleDateFormat(format);
        return sdf.format(date);
    }
}

File: FE/app/src/main/java/com/food/order/utils/DateUtils.kt
package com.food.order.utils

import java.text.SimpleDateFormat
import java.util.Locale

object DateUtils {
    private val apiFormat = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
    private val displayFormat = SimpleDateFormat("dd/MM/yyyy", Locale.getDefault())

    fun formatBirthday(dateString: String?): String {
        if (dateString.isNullOrEmpty()) return "N/A"
        if (dateString.contains("/")) return dateString // Already in display format
        return try {
            val date = apiFormat.parse(dateString)
            if (date != null) displayFormat.format(date) else dateString
        } catch (e: Exception) {
            dateString
        }
    }

    fun toApiDate(displayDate: String?): String? {
        if (displayDate.isNullOrEmpty() || displayDate == "N/A") return null
        if (displayDate.contains("-")) return displayDate // Already in API format
        return try {
            val date = displayFormat.parse(displayDate)
            if (date != null) apiFormat.format(date) else null
        } catch (e: Exception) {
            null
        }
    }
}
